File: python/n171_titleToNum.py
# https://leetcode-cn.com/problems/excel-sheet-column-number/solution/

class Solution:
    def titleToNumber(self, s: str) -> int:
        return sum([(ord(s[i])-64)* 26 ** (len(s)-i-1) for i in range(len(s))])
        
        
if __name__ == "__main__":
    pass

File: cpp/n171_titleToNum.cpp
// https://leetcode-cn.com/problems/excel-sheet-column-number/solution/
#include <iostream>
#include <string>
#include <math.h>
using namespace std;

class Solution{
public:
    int titleToNumber(string s){
        int num = 0;
        for (size_t i = 0; i < s.size(); ++i){
            num = num + (1 + (s[i] - 'A')) * (pow(26, (-1 + s.size() - i)));
        }
        return num;
    }
};

int main(){
    Solution solu;
    string s = "A";
    cout << solu.titleToNumber(s) << endl;
    return 0;
}